authors/apps/articles/tests/test_rate_article.py

```diff
@@ -0,0 +1,170 @@
+from django.urls import reverse
+from ...authentication.models import User
+from ..models import Article
+from rest_framework.test import APIClient
+from rest_framework.test import APITestCase
+from rest_framework import status
+
+
+class TestRateArticle(APITestCase):
+    """ class for testing email verification"""
+
+    def setUp(self):
+        """
+        Prepare test environment for each testcase
+        """
+        self.client = APIClient()
+        self.article = Article()
+        self.signup_url = reverse('authentication:register')
+        self.user_details = {
+            'user': {
+                'username': 'user1',
+                'email': 'evajohnson714@gmail.com',
+                'password': 'somepass12345',
+            }
+        }
+        self.user_details_2 = {
+            'user': {
+                'username': 'user2',
+                'email': 'evajohnson715s@gmail.com',
+                'password': 'somepass12345',
+            }
+        }
+        resp = self.client.post(
+            self.signup_url,
+            self.user_details,
+            format='json')
+        res = self.client.post(
+            self.signup_url,
+            self.user_details_2,
+            format='json')
+        self.token_2 = res.data['token']
+        self.token = resp.data['token']
+        self.email = "test_user@gmail.com"
+        self.name = "test"
+        self.user = User(username=self.name, email=self.email)
+        self.user.set_password("@Winners11")
+        self.user.save()
+        self.user_id = User.objects.get(email=self.email).pk
+        self.slug = "this-is-a-question"
+        title = "this is a question"
+        description = "this is a description"
+        body = "this is a body"
+        author = self.user
+        article = Article(
+            author=author,
+            slug=self.slug,
+            body=body,
+            title=title,
+            description=description)
+        article.save()
+        self.rate_details = {
+            "user": {
+                "slug": self.slug,
+                "rate": 3
+            }
+        }
+        self.rate_url = "http://localhost:8000/api/article/"\
```

Comment:
we should have the staging and production urls as environmental variables.

Reply:
Look at this implementation `https://github.com/andela/ah-orcas/blob/bb63b66d2f6092ced1431dea6d91704690a42576/authors/apps/authentication/views.py#L160`